### ISSUE TITLE
ORM: Make connections optional

### ISSUE THREAD

**Issue by richdunajewski** [balderdashy/sails#3361]
Is there currently a way to make a DB connection optional at sails lift time? Understandably the idea is to catch broken DB connections when the ORM hook loads, but we have a need where we want the application to load and report that the database connection is broken. Under the current scheme, the ORM hook blocks the boostrap process until the hook timeout fires, bringing the server down again.

A better example may be -- we're looking for a system where the application can come online and report periodically to a monitoring service the health of the instance, and the broken DB connection would be one such message we would send to the service. 

Assuming that this is not presently supported, could this be supported in some form in the future? If I could listen for the ORM timeout event, I could send off any notifications that the DB is down, but that would be a one-shot deal. The server would be shut down after that regardless. Is there such an event for hook:orm:timeout or similar?

**Comment by foobar8675** [comment]
i'm also pretty new to sails, but what I'm planning to do on AWS is rely on AWS RDS monitoring as well as EC2 monitoring. Having the monitors be independent sounds a bit safer to me than relying on your application tier reporting db health. So for your example, I would see ec2 (sails) is down and look at AWS reporting to see if something is whacky with an RDS instance.  not an answer to your question by any means - just a different thought.

**Comment by sailsbot** [comment]
Thanks for posting, @richdunajewski.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
